### PR TITLE
[CI] Only build fuzzer images when necessary

### PR DESCRIPTION
**Pull request by jonathanmetzman**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,34 +14,3 @@ jobs:
     - name: Run presubmit checks
       run: |
         make presubmit
-
```
Comment (inferno-chromium) [inline .github/workflows/ci.yml]: maybe call this file presubmit.yml as CI name does not make sense. all stuff in this dir is CI ?

File: fuzzers/test [hunk 2]
```diff
@@ -0,0 +1 @@
+hello
```
Comment (inferno-chromium) [inline fuzzers/test]: need to remember to remove this. just a test i think :)

**Comment by jonathanmetzman** [comment]
This change isn't 100% necessary and may actually cause CI not to run when it should if I implemented it incorrectly. But I like it because I'd rather not see so many fuzzer builds unless a fuzzer is actually being changed. WDYT? https://github.com/google/fuzzbench/actions/runs/50224004 is proof that it works when a fuzzer is changed.

**Comment by inferno-chromium** [comment]
> This change isn't 100% necessary and may actually cause CI not to run when it should if I implemented it incorrectly. But I like it because I'd rather not see so many fuzzer builds unless a fuzzer is actually being changed. WDYT? https://github.com/google/fuzzbench/actions/runs/50224004 is proof that it works when a fuzzer is changed.

That is file, might want to think of running that integration test you had in presubmit since that can make sure workflow does not break. i agree we shouldn't be building fuzzers, benchmarks for all cases, but that integration test can catch obvious breakages.

File: fuzzers/test (hunk 2 above)
Comment (jonathanmetzman) [reply fuzzers/test]: Yeah I did.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jonathanmetzman) [reply .github/workflows/ci.yml]: Done.

**Pull request merged**